tests/testthat/test-write.R

```diff
@@ -82,129 +71,111 @@ test_that("xportr_write: expect error when invalid multibyte string is passed in
 })
 
 test_that("xportr_write: expect error when file name is over 8 characters long", {
-  tmpdir <- tempdir()
-  tmp <- file.path(tmpdir, paste0(paste(letters[1:9], collapse = ""), ".xpt"))
-
-  on.exit(unlink(tmpdir))
-
-  expect_error(xportr_write(data_to_save, tmp))
+  skip_if_not_installed("withr")
+  expect_error(
+    xportr_write(
+      data_to_save(),
+      withr::local_file(paste0(paste(letters[1:9], collapse = ""), ".xpt"))
+    ),
+    "`\\.df` file name must be 8 characters or less\\."
+  )
 })
 
 test_that("xportr_write: expect error when file name contains non-ASCII symbols or special characters", {
-  tmpdir <- tempdir()
-  tmp <- file.path(tmpdir, "<test>.xpt")
-
-  on.exit(unlink(tmpdir))
-
-  expect_error(xportr_write(data_to_save, tmp))
+  skip_if_not_installed("withr")
+  expect_error(
+    xportr_write(data_to_save(), withr::local_file("<test>.xpt"), strict_checks = TRUE),
+    "`\\.df` cannot contain any non-ASCII, symbol or underscore characters\\."
+  )
 })
 
 test_that("xportr_write: expect warning when file name contains underscore and strict_checks = FALSE", {
-  tmpdir <- tempdir()
-  tmp <- file.path(tmpdir, "test_.xpt")
-
-  on.exit(unlink(tmpdir))
-
-  expect_warning(xportr_write(data_to_save, tmp, strict_checks = FALSE))
+  skip_if_not_installed("withr")
+  expect_warning(
+    xportr_write(data_to_save(), withr::local_file("test_.xpt"), strict_checks = FALSE),
+    "`\\.df` cannot contain any non-ASCII, symbol or underscore characters\\."
+  )
 })
 
 test_that("xportr_write: expect error when label contains non-ASCII symbols or special characters", {
-  tmpdir <- tempdir()
-  tmp <- file.path(tmpdir, "xyz.xpt")
-
-  on.exit(unlink(tmpdir))
-
+  skip_if_not_installed("withr")
   expect_error(
     xportr_write(
-      data_to_save,
-      tmp,
-      expect_error(
-        xportr_write(
-          data_to_save,
-          domain = "data_to_save",
-          tmp,
-          metadata = data.frame(
-            dataset = "data_to_save",
-            label = "çtestç"
-          )
-        )
+      data_to_save(),
+      domain = "data_to_save",
+      path = withr::local_file("xyz.xpt"),
+      metadata = data.frame(
+        dataset = "data_to_save",
+        label = "çtestç"
       )
-    )
+    ),
+    "`label` cannot contain any non-ASCII, symbol or special characters"
```

Comment:
ℹ️ Problem detected with missing test coverage. (recursive call of `xportr_write`)
ℹ️ Adds exact error that is expected (test was passing, but outputing the wrong error)

note: I can separate this from this PR, just let me know if this is too out of scope

Reply:
@EeethB was this the same as the testing issue in the other PR - with the lack of coverage??  or separate issue? IF not, @averissimo I think this can be handled in this PR.  

I like the word graceful in the PR titile